src/main/java/Dul.java

```diff
@@ -0,0 +1,251 @@
+                } else {
+                    addDeadlineTask(command[1]);
+                }
+                break;
+            case "event":
+                addEventTask(command[1]);
+                break;
+            case "delete":
+                deleteTask(Integer.parseInt(command[1]) - 1);
+                break;
+            case "bye":
+                break;
+            default:
+                System.out.println("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    public static void listTasks() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println((i + 1) + "." + tasks.get(i).toString());
+        }
+    }
+
+    public static void addTodoTask(String taskType) {
+        tasks.add(new TodoTask(taskType));
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + tasks.get(tasks.size() - 1).toString());
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    public static void addDeadlineTask(String taskType) {
+        String[] parts = taskType.split(" /by ", 2);
+        String description = parts[0];
+        String by = parts[1];
+        tasks.add(new DeadlineTask(description, by));
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + tasks.get(tasks.size() - 1).toString());
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    public static void addEventTask(String taskType) {
+        String[] parts = taskType.split(" /from ", 2);
+        String[] eventParts = parts[1].split(" /to ", 2);
+        String description = parts[0];
+        String from = eventParts[0];
+        String to = eventParts[1];
+        tasks.add(new EventTask(description, from, to));
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + tasks.get(tasks.size() - 1).toString());
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    public static void markTaskDone(int index) {
+        tasks.get(index).markDone();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println("  " + tasks.get(index).toString());
+    }
+
+    public static void markTaskNotDone(int index) {
+        tasks.get(index).markNotDone();
+        System.out.println("OK, I've marked this task as not done yet:");
+        System.out.println("  " + tasks.get(index).toString());
+    }
+
+    public static void deleteTask(int index) {
+        Task removedTask = tasks.remove(index);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println("  " + removedTask.toString());
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+}
```

Comment:
your main class is doing a lot of things. you can refactor the code heavily to make it lean.